File: aurelia/vision/analyze-image.js
"use strict";

const args = process.argv.slice(2);
const VISION_KEY = args[0];
const VISION_ENDPOINT = args[1];

import request from "request";

let key = VISION_KEY;
let endpoint = VISION_ENDPOINT;

var uriBase = endpoint + "vision/v3.1/analyze";

const imageUrl = "D:/Vault/Coding/Repositories/aurelia/data/display.png";

// Request parameters.
const params = {
  visualFeatures: "Categories,Description,Color",
  details: "",
  language: "en",
};

const options = {
  uri: uriBase,
  qs: params,
  body: '{"url": ' + '"' + imageUrl + '"}',
  headers: {
    "Content-Type": "application/octet-stream",
    "Ocp-Apim-Subscription-Key": key,
  },
};

console.log(options);

request.post(options, (error, response, body) => {
  console.log(response);
  if (error) {
    console.log("Error: ", error);
    return;
  }
  let jsonResponse = JSON.stringify(JSON.parse(body), null, "  ");
  console.log("JSON Response\n");
  console.log(jsonResponse);
});

File: aurelia/vision/analyze-image.py
import requests
import json


_key  = "key"
_url = "url"


path_to_file = "D:/Vault/Coding/Repositories/aurelia/data/display.png"
with open(path_to_file, 'rb') as f:
    data = f.read()


# Computer Vision parameters
params = {'mode' : 'Categories,Description,Color'}

headers = dict()
headers['Ocp-Apim-Subscription-Key'] = _key
headers['Content-Type'] = 'application/octet-stream'

json_d = None
response = requests.request( 'post', _url, json = json_d, data = data, headers = headers, params = params )
# Check if the request was successful
if response.status_code == 200:
    result = response.json()

    # Save the result to a JSON file
    output_file_path = "../data/result.json"
    with open(output_file_path, 'w', encoding='utf-8') as json_file:
        json.dump(result, json_file, ensure_ascii=False, indent=4)
    
    print(f"Result saved to {output_file_path}")
else:
    # Print error information if the request was not successful
    print(f"Error code: {response.status_code}")
    print(f"Message: {response.json()}")



